inception/inception-api-annotation-experimental/src/main/ts/client/AnnotationExperienceAPIImpl.ts

```diff
@@ -16,293 +16,448 @@
+            that._stompClient.subscribe("/queue/new_document_for_client/" + that._clientName, function (msg) {
+                that.onNewDocument(Object.assign(new NewDocumentResponse(), JSON.parse(msg.body)));
             }, {id: "new_document"});
 
-            that.stompClient.subscribe("/queue/new_viewport_for_client/" + that.client, function (msg) {
-                that.receiveNewViewportMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            that._stompClient.subscribe("/queue/new_viewport_for_client/" + that._clientName, function (msg) {
+                that.onNewViewport(Object.assign(new NewViewportResponse(), JSON.parse(msg.body)));
             }, {id: "new_viewport"});
 
-            that.stompClient.subscribe("/queue/selected_annotation_for_client/" + that.client, function (msg) {
-                that.receiveSelectedAnnotationMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            that._stompClient.subscribe("/queue/selected_annotation_for_client/" + that._clientName, function (msg) {
+                that.onSpanSelect(Object.assign(new SelectSpanResponse(), JSON.parse(msg.body)));
             }, {id: "selected_annotation"});
 
-            that.stompClient.subscribe("/queue/error_for_client/" + that.client, function (msg) {
-                that.receiveErrorMessageByServer(Object.assign(new ServerMessage(), JSON.parse(msg.body)));
+            that._stompClient.subscribe("/queue/selected_relation_for_client/" + that._clientName, function (msg) {
+                that.onRelationSelect(Object.assign(new SelectRelationResponse(), JSON.parse(msg.body)));
+            }, {id: "selected_relation"});
+
+            that._stompClient.subscribe("/queue/error_for_client/" + that._clientName, function (msg) {
+                that.onError(Object.assign(new ErrorMessage(), JSON.parse(msg.body)));
+            }, {id: "error_message"});
+
+            that._stompClient.subscribe("/queue/all_spans_for_client/" + that._clientName, function (msg) {
+                that.onAllSpans(Object.assign(new AllSpanResponse(), JSON.parse(msg.body)));
+            }, {id: "all_spans"});
+
+            that._stompClient.subscribe("/queue/all_relations_for_client/" + that._clientName, function (msg) {
+                that.onAllRelations(Object.assign(new AllRelationResponse(), JSON.parse(msg.body)));
             }, {id: "error_message"});
         };
 
 
         // ------ STOMP ERROR HANDLING ------ //
 
-        this.stompClient.onStompError = function (frame) {
+        this._stompClient.onStompError = function (frame) {
             console.log('Broker reported error: ' + frame.headers['message']);
             console.log('Additional details: ' + frame.body);
         };
 
-        this.stompClient.activate();
+        this._stompClient.activate();
+    }
+
+    multipleSubscriptions()
```

Comment:
What is the purpose of this method?

Reply:
Whenever we switch a document / viewport of a document, we need to subscribe to the new channels for everything that is shown. Otherwise we would not get any updates on the current viewport.
This will also be explained in documentation PR as comment: